src/microbe_stage/PatchManager.cs

```diff
@@ -145,9 +143,24 @@ public void UpdateAllPatchLightLevels()
         if (!gameWorld.WorldSettings.DayNightCycleEnabled)
             return;
 
-        var multiplier = gameWorld.LightCycle.DayLightFraction;
-        compoundCloudSystem.SetBrightnessModifier(multiplier * (compoundCloudBrightness - 1.0f) + 1.0f);
         gameWorld.UpdateGlobalLightLevels();
+
+        var maxLightLevel = biome.Conditions.GetCompound(Compound.Sunlight, CompoundAmountType.Biome).Ambient;
+
+        float multiplier;
+
+        if (maxLightLevel > 0.0f)
+        {
+            multiplier = biome.Conditions.GetCompound(Compound.Sunlight, CompoundAmountType.Current).Ambient *
+                CurrentGame!.GameWorld.LightCycle.DayLightFraction;
```

Comment:
Doesn't this end up applying the light level twice? Because the current sunlight compound should already follow the light cycle, and here that is then further multiplied by data from the LightCycle.